### PR TITLE
Do not polyfill node bindings by default

### PR DESCRIPTION
**Pull request by ooflorent**
**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

yes

**Does this PR introduce a breaking change?**

yes

**What needs to be documented once your changes are merged?**

1. `node` has fewer options:
    - `global`
    - `__filename`
    - `__dirname`

2. `Buffer`, `setImmediate`, `process` and `console` are no longer polyfilled
3. `global` is still supported but is disabled by default
4. `require("vm")` or other node builtin packages are no longer polyfilled.  
Use `resolve.alias` instead.

### CODE REVIEW

**Review by sokra — APPROVED** [review]

**Review by sokra — CHANGES_REQUESTED** [review]
For improved migration we need a note in the error messages when a node.js module fails resolving.

File: lib/dependencies/GlobalDependency.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+*/
+
+"use strict";
+
+const RuntimeGlobals = require("../RuntimeGlobals");
+const ConstDependency = require("./ConstDependency");
+
+exports.createGlobalDependency = range => {
```
Comment (sokra) [inline lib/dependencies/GlobalDependency.js]: Inline this code and remove the file.

File: test/StatsTestCases.test.js [hunk 2]
```diff
@@ -131,7 +131,7 @@ describe("StatsTestCases", () => {
 				actual = actual
 					.replace(/\r\n?/g, "\n")
 					.replace(/[\t ]*Version:.+\n/g, "")
-					.replace(path.join(base, testName), "Xdir/" + testName)
+					.replace(new RegExp(path.join(base, testName), "g"), "Xdir/" + testName)
```
Comment (sokra) [inline test/StatsTestCases.test.js]: This doesn't work for windows paths and paths containing chars that need to be excaped.

**Review by sokra — CHANGES_REQUESTED** [review]

**Comment by ooflorent** [comment]
Please squash-merge, the history is lame.

**Comment by ooflorent** [comment]
History is now clean

**Comment by TheKidCoder** [comment]
@claude review

**Comment by TheKidCoder** [comment]
@claude review

**Comment by TheKidCoder** [comment]
@claude review

**Comment by TheKidCoder** [comment]
@claude review

**Comment by TheKidCoder** [comment]
@claude review

**Pull request merged**